lib/Parser/Parse.cpp

```diff
@@ -3624,6 +3624,10 @@ ParseNodePtr Parser::ParseTerm(BOOL fAllowCall,
     case tkIMPORT:
         if (m_scriptContext->GetConfig()->IsES6ModuleEnabled() && m_scriptContext->GetConfig()->IsESDynamicImportEnabled())
         {
+            if (!fAllowCall)
+            {
+                Error(ERRTokenAfter, _u("import"), _u("new"));
```

Comment:
Is this supposed to be a SyntaxError? I would expect this is a runtime error but (full disclosure) I didn't look closely at the spec.

Reply:
Syntax error per test262: https://github.com/tc39/test262/blob/master/test/language/expressions/dynamic-import/syntax/invalid/top-level-no-new-call-expression.js

Reply:
Seems right. I should have taken a look at the proposal. Appears dynamic import is a special type of CallExpression and can't be a NewExpression. Similar to SuperCall (which also throws SyntaxError when called with new). At least, this is according to https://tc39.github.io/proposal-dynamic-import/#prod-ImportCall

Thanks for confirming.

Reply:
For what it’s worth I was surprised this was a syntax error too.  `import(“”) += 1` is *not* a syntax error (it’s a runtime error though), but this is.  The spec is weird sometimes.

Reply:
import('') +=1 is meant to be an early error, though a Reference error not syntax error but it's not at runtime. Note currently CC does a Syntax error there's an open issue for it.